src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -286,31 +286,30 @@ private DataAccessException getFaultToInsert(LockType lockType) {
     public UUID createAndLock(Dataset dataset, String flightId) throws IOException, SQLException {
         logger.debug("Lock Operation: createAndLock datasetId: {} for flightId: {}", dataset.getId(), flightId);
         String sql = "INSERT INTO dataset " +
-            "(name, default_profile_id, flightid, description, sharedlock) " +
-            "VALUES (:name, :default_profile_id, :flightid, :description, ARRAY[]::TEXT[]) ";
+            "(name, default_profile_id, id, flightid, description, sharedlock) " +
+            "VALUES (:name, :default_profile_id, :id, :flightid, :description, ARRAY[]::TEXT[]) ";
 
         MapSqlParameterSource params = new MapSqlParameterSource()
             .addValue("name", dataset.getName())
             .addValue("default_profile_id", dataset.getDefaultProfileId())
+            .addValue("id", dataset.getId())
             .addValue("flightid", flightId)
             .addValue("description", dataset.getDescription());
         DaoKeyHolder keyHolder = new DaoKeyHolder();
         try {
             jdbcTemplate.update(sql, params, keyHolder);
         } catch (DuplicateKeyException dkEx) {
-            throw new InvalidDatasetException("Dataset name already exists: " + dataset.getName(), dkEx);
-        }
+            throw new InvalidDatasetException(
+                "Dataset name or id already exists: " + dataset.getName() + ", " + dataset.getId(), dkEx);
 
-        UUID datasetId = keyHolder.getId();
-        dataset.id(datasetId);
-        dataset.createdDate(keyHolder.getCreatedDate());
```

Comment:
Is it no longer necessary to populate `dataset.createdDate` generated by the db insert operation?